### DISCUSSION TITLE
MAUI - IOS - Scrolling not working on Pages

### DISCUSSION THREAD

**Discussion by Riffy135** [dotnet/maui#12399] [Q&A]
Hi

I have an issue with getting CollectionView Pages to scroll in IOS emulators with VS 2022 (v 17.4.3)  MAUI.  It works fine for Android and Windows. Only the first page gets displayed on IOS emulator.

In fact this seems to be an issue with scrolling on all types of ContentPages. eg TableView

Below is a snapshot of my XAML page using CollectionView :

    <RefreshView x:DataType="local:Level1ViewModel" IsRefreshing="{Binding IsRefreshing, Mode=TwoWay}"
                     Command="{Binding RefreshCommand}">
        <CollectionView x:Name="ItemsListView" BackgroundColor="{StaticResource  PageBckColour}"
                        ItemsSource="{Binding Item}"  EmptyView="No Items Found" 
                        SelectionMode="Single">
            <CollectionView.ItemTemplate>
                <DataTemplate>
                        <StackLayout Padding="{x:OnPlatform Android=8, WinUI=5, iOS=10}" x:DataType="model:Item" >
                        <Label Text="{Binding Text}" 
                            LineBreakMode="WordWrap"  
                            Style="{DynamicResource ListItemTextStyle}" 
                            TextColor="{StaticResource TxtColour}"
                            FontAttributes="Bold"
                            FontSize="{x:OnPlatform Android=13, WinUI=14, iOS=14}" ></Label>
                            <Label Text="{Binding Description}"  
                            LineBreakMode="WordWrap"  
                            Style="{DynamicResource ListItemDetailTextStyle}" 
                            TextColor="Blue" 
                            FontSize="{x:OnPlatform Android=12, WinUI=12, iOS=13}"></Label>
                        <StackLayout.GestureRecognizers>
                            <TapGestureRecognizer 
                                NumberOfTapsRequired="1"
                                Command="{Binding Source={RelativeSource AncestorType={x:Type local:Level1ViewModel}}, Path=ItemTapped}"		
                                CommandParameter="{Binding .}">
                            </TapGestureRecognizer>
                        </StackLayout.GestureRecognizers>
                    </StackLayout>
                </DataTemplate>
            </CollectionView.ItemTemplate>
        </CollectionView>
    </RefreshView>
</ContentPage>

Any suggestions as how to  resolve the issue would be welcome.

Thanks

**Comment by AnthonyIacono** [comment]
Try adding VerticalOptions with the value of FillAndExpand for all your collection views. We had to do this when migrating to MAUI or else nothing would scroll on iOS. Probably a bug.

**Reply by Riffy135** [reply]
Hi Anthony

My other pages did have VerticalOptions with the value of FillAndExpand and that too did not make the scroll work. I tried changing the above XAML page to include this and still have the scrolling issue.

I agree with you and think it is a bug with the MAUI.

Thanks

**Reply by AnthonyIacono** [reply]
Did you add the option to the CollectionView itself?

**Reply by Riffy135** [reply]
Yes I did for both CollectionView and Refreshview and for one without the other to cover all bases.

**Reply by Riffy135** [reply]
The scrolling issue also occurs with a new project (selecting a MAUI Template) with a ContentPage which has a simple page with more then one pageful of controls eg Labels.

Is anyone else experiencing this?

Thanks

**Reply by vikher** [reply]
IM facing the same problem targeting iOS 16.2 , actually adding Fill and Expand to the collection View solves this problem, but I can't wrap up the collection view in a refresh view because the bug starts occuring. This is a sample collection view that worked with this work around :

            <CollectionView
                x:Name="mywashersview"
                EmptyView="No Washing Machines to display"
                ItemsSource="{Binding WashersList}"
                SelectedItem="{Binding SelectedMachine}"
                SelectionChangedCommand="{Binding SelectedCommand}"
                SelectionChangedCommandParameter="{Binding SelectedItem, Source={x:Reference mywashersview}}"
                SelectionMode="Single"
                VerticalOptions="FillAndExpand">
            </CollectionView>
